test/integration/deployment/deployment_test.go

```diff
@@ -807,34 +805,34 @@ func TestScaledRolloutDeployment(t *testing.T) {
 		update.Spec.Template.Spec.Containers[0].Image = fakeImage2
 	})
 	if err != nil {
-		t.Fatalf("failed updating deployment %q: %v", deploymentName, err)
+		t.Fatalf("failed updating deployment %q: %v", name, err)
 	}
 	if err = tester.waitForDeploymentRevisionAndImage("2", fakeImage2); err != nil {
 		t.Fatal(err)
 	}
 
 	// Verify the deployment has minimum available replicas after 2nd rollout
 	minAvailableReplicas := deploymentutil.MinAvailable(tester.deployment)
-	if err = tester.waitForDeploymentHasMinimumAvailableReplicas(minAvailableReplicas); err != nil {
-		t.Fatalf("failed waiting deployment %q to have minimum number of available replicas: %v", deploymentName, err)
+	if tester.deployment.Status.AvailableReplicas < minAvailableReplicas {
+		t.Fatalf("deployment %q does not have minimum number of available replicas after 2nd rollout", name)
 	}
 
-	// Verify old replicaset of 1st rollout still has desired replicas (it is not scaled down due to new replicaset is blocked)
+	// Wait for old replicaset of 1st rollout to have desired replicas (it is not scaled down due to new replicaset is blocked)
 	firstRS, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).Get(firstRS.Name, metav1.GetOptions{})
 	if err != nil {
 		t.Fatalf("failed to get replicaset %q: %v", firstRS.Name, err)
 	}
-	if err = tester.waitForReplicaSetHasDesiredReplicas(firstRS); err != nil {
-		t.Fatalf("failed waiting replicaset %q to have desired number of replicas: %v", firstRS.Name, err)
+	if err = tester.waitRSStable(firstRS); err != nil {
+		t.Fatal(err)
```

Comment:
@janetkuo FYI, the error returned has all the required information, so `t.Fatal(err)` suffices.